receiver/otelarrowreceiver/otelarrow_test.go

```diff
@@ -351,23 +348,38 @@ func TestOTelArrowShutdown(t *testing.T) {
 			}()
 
 			start := time.Now()
-			var once sync.Once
-
-			// Send traces to the receiver until we signal via done channel, and then
-			// send one more trace after that.
-			go generateTraces(func(td ptrace.Traces) {
-				if time.Since(start) > 5*time.Second {
-					once.Do(func() {
-						if cooperative {
-							require.NoError(t, stream.CloseSend())
-						}
-					})
-					return
+
+			// Send traces to the receiver until we signal.
+			go func() {
+				for time.Since(start) < 5*time.Second {
+					td := testdata.GenerateTraces(1)
+					batch, batchErr := producer.BatchArrowRecordsFromTraces(td)
+					require.NoError(t, batchErr)
+					require.NoError(t, stream.Send(batch))
 				}
-				batch, batchErr := producer.BatchArrowRecordsFromTraces(td)
-				require.NoError(t, batchErr)
-				require.NoError(t, stream.Send(batch))
-			}, doneSignalGrpc)
+
+				if cooperative {
+					require.NoError(t, stream.CloseSend())
+				}
+			}()
```

Comment:
Note that this is a simplification of the former call to `generateTraces()` which is really written for another test in this source file. The resulting simplification is easier to read.